cache/network/download.go

```diff
@@ -49,17 +49,20 @@ func Download(ctx context.Context, params DownloadParams, logger log.Logger) (ma
 	}
 
 	logger.Debugf("Download archive")
-
-	downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
-	if downloadErr != nil {
-		return "", fmt.Errorf("failed to download archive: %w", downloadErr)
+	err = downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to download archive: %w", err)
 	}
 
 	return restoreResponse.MatchedKey, nil
 }
 
-func customRetryFunction(ctx context.Context, resp *http.Response, err error) (bool, error) {
-	return retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+func createCustomRetryFunction(logger log.Logger) func(context.Context, *http.Response, error) (bool, error) {
+	return func(ctx context.Context, resp *http.Response, downloadErr error) (bool, error) {
+		retry, err := retryablehttp.DefaultRetryPolicy(ctx, resp, downloadErr)
+		logger.Debugf("CheckRetry: retry=%v ; err=%+v ; downloadErr=%+v", retry, err, downloadErr)
```

Comment:
I would consider making this a warning instead of a debug log. The reason is that customers have to explicitly turn on debug logging for the step to see this. If it is a warning then it will be logged for every customer who runs into this issue. And then we could easily spot the various errors without asking every one of them to turn on the debug logging.